google-cloud-firestore/lib/google/cloud/firestore/convert.rb

```diff
@@ -140,47 +146,37 @@ def hash_to_geo_point hash, pairs = nil
             raise ArgumentError, "value is not a geo point" unless pairs
 
             Google::Type::LatLng.new(
-              latitude: pairs.first.last,
-              longitude: pairs.last.last,
+              latitude:  pairs.first.last,
+              longitude: pairs.last.last
             )
           end
 
-          def writes_for_create doc_path, data
-            writes = []
-
-            if is_field_value_nested data, :delete
+          def write_for_create doc_path, data
+            if field_value_nested? data, :delete
               raise ArgumentError, "DELETE not allowed on create"
             end
             raise ArgumentError, "data is required" unless data.is_a? Hash
 
             data, field_paths_and_values = remove_field_value_from data
 
-            if data.any? || field_paths_and_values.empty?
-              write = Google::Cloud::Firestore::V1::Write.new(
-                update: Google::Cloud::Firestore::V1::Document.new(
-                  name: doc_path,
-                  fields: hash_to_fields(data)),
-                current_document: Google::Cloud::Firestore::V1::Precondition.new(
-                  exists: false)
-              )
-              writes << write
-            end
-
-            if field_paths_and_values.any?
-              transform_write = transform_write doc_path, field_paths_and_values
-
-              if data.empty?
-                transform_write.current_document = \
-                  Google::Cloud::Firestore::V1::Precondition.new(exists: false)
-              end
-
-              writes << transform_write
-            end
+            Google::Cloud::Firestore::V1::Write.new(
+              update:            Google::Cloud::Firestore::V1::Document.new(
```

Comment:
Rubocop's rules on lining up hash values makes this look awkward. (The keyword args for the inner Document.new are indented less than the keyword args for the outer Write.new). We can make this more readable by creating the Document (and Precondition) first, assign to temporary instance variables, and then pass those to Write.new.

There are also a few additional cases of this below.

Reply:
Fixed, thank you.